src/device.c

```diff
@@ -176,24 +167,20 @@ swap_char(unsigned char *a, unsigned char *b)
 }
 
 static void
-decode_buf(char result[8], unsigned char buf[8], unsigned char magic_table[8])
+decode_buf(co2mon_data_t result, co2mon_data_t buf, co2mon_magic_table_t magic_table)
 {
-    const unsigned char magic_word[8] = "Htemp99e";
-
-    int i;
-    unsigned char tmp;
-
     swap_char(&buf[0], &buf[2]);
     swap_char(&buf[1], &buf[4]);
     swap_char(&buf[3], &buf[7]);
     swap_char(&buf[5], &buf[6]);
 
-    for (i = 0; i < 8; ++i)
+    for (int i = 0; i < 8; ++i)
     {
         buf[i] ^= magic_table[i];
+        // TODO: do we really need to xor it with magic_table? Its value is {0}
```

Comment:
Yes, we do. You may use any values in magic_table. And the original software calculates magic_table using current time.